.gitlab-ci.yml

```diff
@@ -0,0 +1,268 @@
+    - source /etc/profile.d/ccache.sh
+    - export CCACHE_DIR="$(pwd)/.ccache"; ccache -s
+    - yum-builddep -y mariadb-server
+    - mkdir builddir; cd builddir
+    - cmake -DRPM=$CI_JOB_NAME $CMAKE_FLAGS .. 2>&1 | tee -a ../build-$CI_JOB_NAME-$CI_COMMIT_REF_SLUG.log
+    - eatmydata make package 2>&1 | tee -a ../build-$CI_JOB_NAME-$CI_COMMIT_REF_SLUG.log
+    # @TODO: Don't use -j on Gitlab.com as builds just get stuck when running
+    # multi-proc, needs more debugging
+    - make test || true  # Unit tests constantly fail, see https://jira.mariadb.org/browse/MDEV-25820
+    # - make test-force || true  # mysql-test-runner takes too long, run it in a separate job instead
+    - *rpm_listfiles
+    - mkdir ../rpm; mv *.rpm ../rpm
+    - ccache -s
+  artifacts:
+    when: always  # Must be able to see logs
+    paths:
+      - build-$CI_JOB_NAME-$CI_COMMIT_REF_SLUG.log
+      - rpmlist-$CI_JOB_NAME-$CI_COMMIT_REF_SLUG.log
+      - rpm
+      - builddir/_CPack_Packages/Linux/RPM/SPECS/
+  cache:
+    key: $MARIADB_MAJOR_VERSION-$CI_JOB_NAME
+    paths:
+      - .ccache
+
+centos7:
+  stage: build
+  image: centos:7
+  variables:
+    GIT_STRATEGY: fetch
+    GIT_SUBMODULE_STRATEGY: normal
+  script:
+    - yum install -y yum-utils rpm-build gcc gcc-c++ bison libxml2-devel libevent-devel openssl-devel
+    - yum-builddep -y mariadb-server
+    - mkdir builddir; cd builddir
+    - cmake -DRPM=$CI_JOB_NAME $CMAKE_FLAGS .. 2>&1 | tee -a ../build-$CI_JOB_NAME-$CI_COMMIT_REF_SLUG.log
+    - make package 2>&1 | tee -a ../build-$CI_JOB_NAME-$CI_COMMIT_REF_SLUG.log
+    # @TODO: Don't use -j on Gitlab.com as builds just get stuck when running
+    # multi-proc, needs more debugging
+    - make test || true  # Unit tests constantly fail, see https://jira.mariadb.org/browse/MDEV-25820
+    # - make test-force || true  # mysql-test-runner takes too long, run it in a separate job instead
+    - *rpm_listfiles
+    - mkdir ../rpm; mv *.rpm ../rpm
+  artifacts:
+    when: always  # Must be able to see logs
+    paths:
+      - build-$CI_JOB_NAME-$CI_COMMIT_REF_SLUG.log
+      - rpmlist-$CI_JOB_NAME-$CI_COMMIT_REF_SLUG.log
+      - rpm
+      - builddir/_CPack_Packages/Linux/RPM/SPECS/
+
+mysql-test-run:
+  stage: test
+  dependencies:
+    - fedora
+  script:
+    # Install packages so tests and the dependencies install
+    # @TODO: RPM missing 'patch' as dependency, so installing it manually for now
+    - yum install -y rpm/*.rpm patch
+    # @TODO: Fix on packaging level that /usr/share/mariadb would work and errormsg.sys be found
+    - rm -rf /usr/share/mariadb; ln -s /usr/share/mysql /usr/share/mariadb
+    # mtr expects to be launched in-place and with write access to it's own directories
+    # @TODO: debug test environment
+    - nproc; cat /proc/cpuinfo | grep -F "model name" -m 1; head -n 1 /proc/meminfo; df -h
+    - whoami; id;
+    - cd /usr/share/mysql-test
+    - |
+      echo "
+      main.mysqldump : flaky on Gitlab-CI
```

Comment:
Could [MDEV-23097](https://jira.mariadb.org/browse/MDEV-23097) explain the flakiness of `main.mysqldump`? If yes, maybe the line should mention the ticket?

Reply:
Not sure, I didn't dig deep into it, just to the level to get the pipeline pass.

Reply:
These are still failing: `Failing test(s): main.mysqldump main.flush_logs_not_windows main.mysql_upgrade_noengine`

I will descope researching and fixing these failures. The pipeline can however help to catch that we don't get *more* failures on main test suite on Fedora/CentOS.

Reply:
@dr-m In Gitlab this is easy to investigate. You just go to the pipeline, open the Tests tab, see the failing tests at the top and click "View results" to see the diff of the result/test to study why it failed:
![image](https://user-images.githubusercontent.com/668724/122635982-7db97780-d0ef-11eb-99fe-82daa744a088.png)